.github/workflows/release.yml

```diff
@@ -0,0 +1,88 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        # linux
+        - os: ubuntu-latest
+          python: "3.12"
+          toxenv: py312
+        - os: ubuntu-latest
+          python: "3.11"
+          toxenv: py311
+        - os: ubuntu-latest
+          python: "3.10"
+          toxenv: py310
+        - os: ubuntu-latest
+          python: "3.9"
+          toxenv: py39
+        - os: ubuntu-latest
+          python: "3.8"
+          toxenv: py38
+        # windows
+        - os: windows-latest
+          python: "3.12"
+          toxenv: py312
+        - os: windows-latest
+          python: "3.11"
+          toxenv: py311
+        - os: windows-latest
+          python: "3.10"
+          toxenv: py310
+        - os: windows-latest
+          python: "3.9"
+          toxenv: py39
+        - os: windows-latest
+          python: "3.8"
+          toxenv: py38
+        # macos
+        - os: macos-latest
+          python: "3.12"
+          toxenv: py312
+        - os: macos-latest
+          python: "3.11"
+          toxenv: py311
+        - os: macos-latest
+          python: "3.10"
+          toxenv: py310
+        - os: macos-latest
+          python: "3.9"
+          toxenv: py39
+        - os: macos-latest
+          python: "3.8"
+          toxenv: py38
+        # misc
+        - os: ubuntu-latest
+          python: "3.11"
+          toxenv: pylama
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install dependencies
+        run: python -m pip install -U build
+
+      - name: Build package
+        run: python -m build    
+
+      - name: Publish package to PyPI
+        if: github.repository == 'mozilla/mozdownload' && github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: ${{ secrets.pypi_user }}
+          repository-url: https://test.pypi.org/legacy/
+          password: ${{ secrets.pypi_password }}
```

Comment:
Validation has to happen via a token and not by using a known user. That will make it more secure and easier to update or revoke permissions.